src/main/bootstrap/try_peer.rs

```diff
@@ -69,7 +70,13 @@ impl<UID: Uid> TryPeer<UID> {
             peer,
             socket,
             request: Some((
-                Message::BootstrapRequest(our_uid, name_hash, ext_reachability, our_pk),
+                Message::BootstrapRequest(
+                    our_uid,
+                    name_hash,
+                    our_global_direct_listeners,
+                    our_role,
```

Comment:
hmm what's the use of supplying our addresses if our role is that of a client ? Wasn't the enum you removed, better ?

Reply:
it makes sure the decision is made by listener and that listener always have data required to do so, if we decide to change it's behavior. Anyway, if that seems very unlikely, I could simply make global addresses an optional variable.

Also, for the reference 
```
enum ExternalReachability {
    Required, NotRequired
}
```
doesn't sound OK to me, cause now we are not asking for ext reachability test, rather we're giving the necessary info to do so.